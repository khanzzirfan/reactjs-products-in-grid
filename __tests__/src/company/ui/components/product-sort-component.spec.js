import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ProductSortComponent from 'src/products/ui/components/product-sort-component';
import SortButtonComponent from 'src/products/ui/components/sort-button-component';

describe("Sort Button Component test", () => {
    let _mockFunc = jest.fn();
    let stubDataProps = {
        onSortClick: _mockFunc,
        currentSort: "size"
    };

    let wrapper = null;
    beforeEach(() => {
        wrapper = shallow(<ProductSortComponent {...stubDataProps} />);
    })
    
    it("component should be defined", () => {
        expect(wrapper).toBeDefined();
    });

    it("component should match to snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it(" should render component with correct props", () => {
        expect(wrapper.instance().props.currentSort).toEqual("size");
    });

    it(" should render component with correct sort label", () => {
        let label = wrapper.find({id:"sort-label"});
        expect(wrapper.find('h3').exists()).toBeTruthy();
        expect(wrapper.find('.text-md-right .text-center').exists()).toBeTruthy();
        expect(label.text()).toEqual("Sort");
    });

    it(" should render component with 3 sort-button-components", () => {
        expect(wrapper.find(SortButtonComponent).exists()).toBeTruthy();
        expect(wrapper.find(SortButtonComponent).length).toEqual(3);
    });

    it(" should pass down child component props with right sort option", () => {
        let firstSortButton = wrapper.find(SortButtonComponent).first();
        expect(firstSortButton.prop('sortOption')).toBe('Size');
        
        let lastSortButton = wrapper.find(SortButtonComponent).last();
        expect(lastSortButton.prop('sortOption')).toBe('Id');
    });

    
    it(" buttons are places inside a class btn group large", () => {
        expect(wrapper.find('.btn-group .btn-group-lg').exists()).toBeTruthy();
    });
});