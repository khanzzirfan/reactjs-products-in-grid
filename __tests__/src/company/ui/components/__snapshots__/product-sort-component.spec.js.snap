// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort Button Component test component should match to snapshot 1`] = `
<div
  className="col-lg-12 col-md-12 col-sm-12 mt-2 mb-2"
>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-sm-2 text-md-right text-center"
    >
      <h3
        className=""
        id="sort-label"
      >
        Sort
      </h3>
    </div>
    <div
      className="col-md-8 col-sm-10 blocks"
    >
      <div
        aria-label="Basic example"
        className="btn-group btn-group-lg"
        role="group"
      >
        <SortButtonComponent
          currentSort="size"
          onSortClick={[Function]}
          sortOption="Size"
        />
        <SortButtonComponent
          currentSort="size"
          onSortClick={[Function]}
          sortOption="Price"
        />
        <SortButtonComponent
          currentSort="size"
          onSortClick={[Function]}
          sortOption="Id"
        />
      </div>
    </div>
  </div>
</div>
`;
